File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_13_chain_of_responsibilities/java/_02_after/RequestHandler.java
package com.example._03_behavioral_patterns._13_chain_of_responsibilities.java._02_after;

import com.example._03_behavioral_patterns._13_chain_of_responsibilities.java._01_before.Request;

public abstract class RequestHandler {

    private final RequestHandler nextHandler;

    public RequestHandler(RequestHandler nextHandler) {
        this.nextHandler = nextHandler;
    }

    public void handle(Request request) {
        if (nextHandler != null) {
            nextHandler.handle(request);
        }
    }
}

File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_13_chain_of_responsibilities/kt/_02_after/RequestHandler.kt
package com.example._03_behavioral_patterns._13_chain_of_responsibilities.kt._02_after

import com.example._03_behavioral_patterns._13_chain_of_responsibilities.kt._01_before.Request

abstract class RequestHandler(
    private val nextHandler: RequestHandler?
) {
    open fun handle(request: Request) {
        nextHandler?.handle(request)
    }
}
